crates/polars-core/src/chunked_array/logical/date.rs

```diff
@@ -44,9 +44,9 @@ impl LogicalType for DateChunked {
                     .into_series())
             },
             #[cfg(feature = "dtype-time")]
-            (Date, Time) => Ok(Int64Chunked::full(self.name(), 0i64, self.len())
-                .into_time()
-                .into_series()),
+            Time => {
```

Comment:
Can you remove this in `polars-arrow`? That's the place where we can raise. 

Another benefit is that we then have to  compile less code. ;)

Reply:
It already raises in `polars-arrow`. The problem is that all logical casts fall back to casting their physical representation, so we have to pre-empt the fallback here.

We can also get rid of that fallback (probably a good idea), but that would require a bigger refactor I think.

Reply:
Oh, yeap, you're right. Ignore what I said.